### PR TITLE
Fullscreen plugin

### PR DESCRIPTION
**Pull request by BaileyBrightman**
Tasks:
- [x] Documentation
- [x] Unit Tests
- [x] Typescript Typings
- [x] `npm run build:full`

### CODE REVIEW

**Comment by 902seanryan** [comment]
One quick note:
Just add `.DS_Store` to the `.gitignore` file

I thought I'd done that already but I guess I missed it in this repo. Not a comment on your PR just something we should add.

File: src/Application.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const completionPercentage = 'completionPercentage';
 const speedScale = 'speedScale';
 const timersScale = 'timersScale';
 const inputCount = 'inputCount';
+const fullscreen = 'fullscreen';
```
Comment (902seanryan) [inline src/Application.js:29]: I would update this to be `fullScreen` not `fullscreen` just a little more readable. Make sure it update it everywhere.

**Review by 902seanryan — CHANGES_REQUESTED** [review]

**Review by 902seanryan — APPROVED** [review]

**Review by aberkie — APPROVED** [review]

**Review by 902seanryan — CHANGES_REQUESTED** [review]
Looks like some of the IndexDB stuff made it into this PR, can you remove them and we'll include them in a separate PR once its closer to completion.

**Review by 902seanryan — DISMISSED** [review]

**Comment by 902seanryan** [comment]
So this looks good to me, we just have to hold off merging until we sort out the node version support stuff.

**Review by 902seanryan — DISMISSED** [review]

**Review by aberkie — APPROVED** [review]

**Review by erinjoost — APPROVED** [review]

**Review by deycorinne — APPROVED** [review]
looks good to me-- my one note is that the actual .DS_Store file is still in this PR even though you added it to he .gitignore. A quick `git rm .DS_Store` should fix that

**Review by deycorinne — APPROVED** [review]

**Comment by 902seanryan** [comment]
@BaileyBrightman looks like there's a couple conflicts showing up. If you re-merge develop and then re-build, you should be good to go

**Review by 902seanryan — APPROVED** [review]

**Review by deycorinne — APPROVED** [review]

**Review by aberkie — APPROVED** [review]

**Pull request merged**